### DISCUSSION TITLE
How can i use this package with $keyType = 'string' in my user model.

### DISCUSSION THREAD

**Discussion by felix-exon** [coderflexx/laravel-ticket#61] [Q&A]
I am using keycloak as a auth provider which results in my user model having a string key.
This seems not to be possible easily as i have to manually alter the migrations.
Why is this not supported out of the box ?

**Comment by ousid** [comment]
@felix-exon 

Can you give me an example, on how you're implementing it in your app?

I didn't experience working with this type of system, but I think it's a Laravel authentication related, no?

**Comment by felix-exon** [comment]
Hi @ousid,

yes, it is just a custom Laravel Socialite Provider (https://github.com/SocialiteProviders/Keycloak). Keycloak uses a guid as the key.

Simply I add `$keyType = 'string';` to my User Model.

It seems to work if I change your migrations manually to

```php
...
$table->foreignIdFor(User::class)
...
```

Maybe this could be solved by specifying the User-Model in the config file and then using `foreignIdFor` for that setting.
I made it work, but it was some manual effort.

**Reply by ousid** [reply]
I see @felix-exon 

Thanks for the details.

I'll try to replicate the logic, to let the devs specify the models out of the box in the config file.

There was a PR for that case (https://github.com/coderflexx/laravel-ticket/pull/40) but there are failed tests, and the creator didn't fix the failed tests, so I had to close the __PR__ as incomplete.

Feel free to open a PR, to match your case, and maybe to help people out with similar situations.

Thanks!
